hls4ml/converters/keras/recurrent.py

```diff
@@ -0,0 +1,38 @@
+import numpy as np
+
+from hls4ml.converters.keras_to_hls import parse_default_keras_layer
+from hls4ml.converters.keras_to_hls import keras_handler
+
+from hls4ml.model.hls_model import Quantizer
+from hls4ml.model.hls_model import IntegerPrecisionType
+
+MAXMULT =  4096
+
+rnn_layers = ['LSTM', 'GRU']
+@keras_handler(*rnn_layers)
+def parse_rnn_layer(keras_layer, input_names, input_shapes, data_reader, config):
+    assert(keras_layer['class_name'] in rnn_layers)
+
+    if keras_layer['class_name'] == 'LSTM':
+        div_factor = 4
+    elif keras_layer['class_name'] == 'GRU':
+        div_factor = 3
+    else:
+        div_factor = 1
+
+    layer = parse_default_keras_layer(keras_layer, input_names)
+
+    weights_shape = data_reader.get_weights_shape(layer['name'], 'kernel')
```

Comment:
Instead of inferring attributes from weights, we should extract the number of hidden units and use that to compute the shapes. Keras uses this (`units`), as well as pytorch and onnx (as `hidden_size` attribute).

Also, other LSTM layer properties should be parsed as well. See the [docs](https://keras.io/api/layers/recurrent_layers/lstm/) for all things that can be set. Even if we don't support them all in the current HLS implementation we should be aware of them so we can provide a message to the user if something is not supported as opposed to silently ignoring them and doing something unintended.

Reply:
I followed the same format used for other layers which were using weight_shape to figure out the shape. I've added all the architecture related properties of the RNN layer to the layer dictionary. Only properties associated with training aren't added.

Reply:
What about adding the parsing of all the other layer properties, could you add that @chaitanyaPaikara ?

Reply:
`return_state`, `go_backwards`, `stateful`, `time_major`, `reset_after`... Preferably the conversion should check the unsupported parameters in `initialize` and raise exceptions in case of configuration we can't handle yet.